cfme/automate/service_dialogs.py

```diff
@@ -95,148 +120,159 @@ def __init__(self, label=None, description=None,
+        else:
+            view.flash.assert_message(
+                'Edit of Dialog "{}" was cancelled by the user'.format(self.label))
 
     def update_element(self, second_element, element_data):
-        navigate_to(self, 'Edit')
-        if self.appliance.version > "5.5":
-            tree = accordion.tree("Dialog")
-        else:
-            tree = Tree("dialog_edit_treebox")
-        tree.click_path(self.label, self.tab_label, self.box_label)
-        self.add_element(second_element)
+        view = navigate_to(self, 'Edit')
+        view.element_tree.click_path(self.label, self.tab_label, self.box_label)
+        self.add_element(view, second_element)
         list_ele = []
         list_ele.append(element_data.get("ele_label"))
         list_ele.append(second_element.get("ele_label"))
-        tree.click_path(self.label, self.tab_label, self.box_label)
+        view.element_tree.click_path(self.label, self.tab_label, self.box_label)
         ele_1 = self.element(list_ele[0])
         ele_2 = self.element(list_ele[1])
         sel.drag_and_drop(ele_1, ele_2)
-        sel.click(form_buttons.save)
-        flash.assert_no_errors()
-
-    def delete(self, cancel=False):
-        navigate_to(self, 'Details')
-        cfg_btn(version.pick({version.LOWEST: "Remove from the VMDB",
-                              '5.7': 'Remove Dialog'}),
-                invokes_alert=True)
-        sel.handle_alert(cancel)
-
-    def element_type(self, each_element):
+        view.save_button.click()
+        view = self.create_view(DetailsDialogView)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+
+    def delete(self):
+        view = navigate_to(self, "Details")
+        view.configuration.item_select(
+            version.pick({
+                version.LOWEST: 'Remove from the VMDB',
+                '5.7': 'Remove Dialog'}),
+            handle_alert=True)
+        view = self.create_view(DialogsView)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+        view.flash.assert_success_message(
+            'Dialog "{}": Delete successful'.format(self.label))
+
+    def element_type(self, view, each_element):
         choose_type = each_element.get("choose_type")
         dynamic_chkbox = each_element.get("dynamic_chkbox")
         if choose_type == "Drop Down List" or choose_type == "Radio Button":
             if not dynamic_chkbox:
-                entry_table.click_cell(header='value', value='<New Entry>')
-                fill(element_form, {'entry_value': "Yes",
-                                    'entry_description': "entry_desc"},
-                    action=element_form.add_entry_button)
+                row = view.entry_table.row(Value='<New Entry>')
+                row.click()
+                view.fill({'entry_value': "Yes",
+                           'entry_description': "entry_desc",
+                           'add_entry_button': ""})
```

Comment:
add entry button is an image . need to click on it after filling data .
If I do view.add_entry_button.click I get Input element does not have click().
Any other way to click on it ?